hypothesis-python/docs/settings.rst

```diff
@@ -71,7 +74,7 @@ with each phase corresponding to a value on the :class:`~hypothesis.Phase` enum:
 3. ``Phase.generate`` controls whether new examples will be generated.
 4. ``Phase.target`` controls whether examples will be mutated for targeting.
 5. ``Phase.shrink`` controls whether examples will be shrunk.
-
+6. ``Phase.explain`` controls whether Hypothesis attempts to explain test failures.
```

Comment:
I would love to see some example of how `explain` result looks like somewhere in the docs. Does it fit there?

Reply:
I was planning on showing it off in a blog post - there's nowhere obvious to put it in the docs, and the exact output format is pretty simple:

```python
from hypothesis import Phase, given, strategies as st

@given(st.integers())
def test_reports_branch_in_test(x):
    if x > 10:
        raise AssertionError  # BUG
```
```
_________________________ test_reports_branch_in_test _________________________
Traceback (most recent call last):
  ...
AssertionError
--------------------------------- Hypothesis ----------------------------------
Falsifying example: test_reports_branch_in_test(
    x=11,
)
Explanation:
    These lines were always and only run by failing examples:
        /path/to/test_file.py:6
```

One consideration in "what do we report" is that this format (usually) allows you to click on terminal output and have the relevant file open to that line in your preferred editor.

The alternative approach of reporting *branches* (source/destination pairs of lines) is only rarely more precise in practice, and much more difficult to explain to non-expert users.  "report branches if there are no reportable lines" would be a nice trick to explore in future, though.